### PR TITLE
migrate web example to typescript

### LINKED ISSUE #358: Move sample web application to typescript
https://github.com/apollographql/apollo-cache-persist/tree/master/examples/web

### PR DESCRIPTION
**Pull request by jaecktec**
[fixes #358]

migrated example app to typescript

### CODE REVIEW

File: examples/web/src/index.tsx [hunk 1]
```diff
@@ -2,17 +2,32 @@ import React from 'react';
...
-import { Query, Mutation, Subscription } from '@apollo/client/react/components';
+import { Query } from '@apollo/client/react/components';
 import gql from 'graphql-tag';
 import { InMemoryCache } from '@apollo/client/core';
 import { CachePersistor } from 'apollo3-cache-persist';
+import { PersistentStorage } from "apollo3-cache-persist/types";
 
 let persistor;
 
+class LocalStoragePersistedStorage implements PersistentStorage<string> {
```
Comment (wtrocki) [inline examples/web/src/index.tsx:13]: This is why we needed to go to typescript. Looks like types are not aligned - Still. 

Hoping to get that over the line in separate PR.

Comment (wtrocki) [reply examples/web/src/index.tsx:13]: Actually, if we have the wrapper in the package that would be cool and work outside the box.

**Review by wtrocki — APPROVED** [review]

File: examples/web/src/index.tsx (hunk 1 above)
Comment (jaecktec) [reply examples/web/src/index.tsx:13]: sure I can move it. Give me till 17:00 UTC+2, lunch-break is over

**Comment by wtrocki** [comment]
@jaecktec If you want to create follow up: migrate to hooks, move interface etc. I will not create any issue for those. Take your time

**Pull request merged**